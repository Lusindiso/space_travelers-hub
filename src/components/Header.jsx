import { Link } from 'react-router-dom';

function NavLink() {
  return (
    <nav id="nav-links">
      <Link to="/Rockets">Rockets</Link>
      <Link to="/Missions">Missions</Link>
      <Link to="/Profile">My Profile</Link>
    </nav>
  );
}

function Header() {
  return (
    <div id="page-header">
      <div className="logo">
        <img alt="SpaceX logo" src="./planet.png" />
        <h1>Space Travelers&#39; Hub</h1>
      </div>
      <NavLink />
    </div>
  );
}

export default Header;
