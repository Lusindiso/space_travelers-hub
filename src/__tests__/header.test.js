import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Header from '../components/Header';

describe('Test for Header components', () => {
  it('Check if the Header section are available', () => {
    const header = renderer.create(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    );
    expect(header).toMatchSnapshot();
  });
});
