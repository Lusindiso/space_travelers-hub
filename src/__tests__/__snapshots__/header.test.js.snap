// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Header components Check if the Header section are available 1`] = `
<div
  id="page-header"
>
  <div
    className="logo"
  >
    <img
      alt="SpaceX logo"
      src="./planet.png"
    />
    <h1>
      Space Travelers' Hub
    </h1>
  </div>
  <nav
    id="nav-links"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "underline",
        }
      }
    >
      Rockets
    </a>
    <a
      className=""
      href="/missions"
      onClick={[Function]}
    >
      Missions
    </a>
    <a
      className=""
      href="/profile"
      onClick={[Function]}
    >
      My Profile
    </a>
  </nav>
</div>
`;
